src/validations/css/stylelint.js

```diff
@@ -0,0 +1,65 @@
+import Validator from '../Validator';
+
+const errorMap = {
+  // CSS Syntax Errors
+  'Unclosed block': (error) => {
+    const token = error.source;
+
+    return {
+      reason: 'missing-closing-curly',
+      payload: {error: token},
+    };
+  },
+
+  // Custom Rules
+  'declaration-block-trailing-semicolon': (error) => {
+    const token = error.css;
+
+    return {
+      reason: 'missing-semicolon',
+      payload: {error: token},
+    };
+  },
+};
+
+class StyleLintValidator extends Validator {
+  constructor(source) {
+    super(source, 'css', errorMap);
+  }
+
+  _getRawErrors() {
+    return System.import('../linters').then(
+      ({stylelint}) => stylelint(
+        this._source
+      ).then(
+        (result) => ({result})
+      ).catch(
+        (syntaxError) => ({syntaxError})
+      )
```

Comment:
Yup. Had to wrap `result` and `syntaxError` as objects because if I passed them like is, they get treated as arrays and each individual attribute gets passed into `_keyForError`.